File: Coding Challenges/Leetcode/1. Two Sum.js
/**
 * @param {number[]} nums
 * @param {number} target
 * @return {number[]}
 */
 var twoSum = function(nums, target) {
    let map = new Map()
    for(let i=0;i<nums.length;i++){
        if(map.has(target-nums[i])) {
          let x = map.get(target-nums[i])
          return [x, i]
        }
        if(!map.has(nums[i])) map.set(nums[i], i)
    }
};

// Runtime: 72 ms, faster than 94.30% of JavaScript online submissions for Two Sum.
// Memory Usage: 38.6 MB, less than 73.70% of JavaScript online submissions for Two Sum.

File: Coding Challenges/Leetcode/1. Two Sum.go
func twoSum(nums []int, target int) []int {
    hmap := make(map[int]int)
    for i, n := range nums {
        if x, y := hmap[target-n]; y{
            return []int{x, i}
        }
        hmap[n] = i
    }
    return []int{}
}
